notebooks/fs.ipynb

```diff
@@ -0,0 +1,1586 @@
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file2.txt -> tmp/copy-keys-target/data/file2-copy.txt: Initial state: Pending\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file2.txt -> tmp/copy-keys-target/data/file2-copy.txt: Checking requirements\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file2.txt -> tmp/copy-keys-target/data/file2-copy.txt: Requirement(s) ready\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file2.txt -> tmp/copy-keys-target/data/file2-copy.txt: Executing\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file2.txt -> tmp/copy-keys-target/data/file2-copy.txt: Final state: Ready\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file3.csv -> tmp/copy-keys-target/data/file3-copy.csv: Initial state: Pending\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file3.csv -> tmp/copy-keys-target/data/file3-copy.csv: Checking requirements\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file3.csv -> tmp/copy-keys-target/data/file3-copy.csv: Requirement(s) ready\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file3.csv -> tmp/copy-keys-target/data/file3-copy.csv: Executing\n",
+      "[2024-08-30T15:21:47]     INFO Copy fixtures/fs/file3.csv -> tmp/copy-keys-target/data/file3-copy.csv: Final state: Ready\n",
+      "[2024-08-30T15:21:47]     INFO File copies: Final state: Ready\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "rmtree(\"tmp/copy-keys-target\", ignore_errors=True)\n",
+    "fs.copy(\n",
+    "    config=\"fixtures/fs/copy-keys-config.yaml\",\n",
+    "    target_dir=\"tmp/copy-keys-target\",\n",
+    "    keys=[\"files\",\"to\",\"copy\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d62ae369-69f9-4003-b5d3-d7b5908f23af",
+   "metadata": {},
+   "source": [
+    "With this information provided, the copy is successful."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "59c67e22-fe98-4e74-8b0b-b40e24a804e8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[01;34mtmp/copy-keys-target\u001b[0m\n",
+      "├── \u001b[01;34mdata\u001b[0m\n",
+      "│   ├── file2-copy.txt\n",
+      "│   └── file3-copy.csv\n",
+      "└── file1-copy.nml\n",
+      "\n",
+      "2 directories, 3 files\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "tree tmp/copy-keys-target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a1adba8-2daf-4fb6-b224-980b134f011c",
+   "metadata": {},
+   "source": [
+    "### Using the `Copier` class\n",
+    "\n",
+    "An alternative to using `copy()` is using the `Copier` class to stage files to copy and carry out the copy with `Copier.go()`."
```

Comment:
```suggestion
    "An alternative to using `copy()` is to instantiate a `Copier` object , then call its `go()` method."
```
- To "stage", the the parlance of these modes, is to actually put the things -- copies, links, directories -- so I feel like it's `go()` that does the staging. I tried here to separate that term from the act of just instantiating the `Copier` object, which doesn't make any changes yet.
- This suggested change eliminated `carry out` from the sentence, but if it had remained it should have become `carrying out` to agree with the two instances `using`: `using... using... carrying...` rather than `using... using... carry`.